File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        // Array was not rotated or have only one element.
        if (nums.front() <= nums.back()) return nums.front();
        
        int n = nums.size(),
            l = 0,
            r = n - 1;
        
        while (l < r) {
            int m = l + (r - l)/2,  
                mid = nums[m];
            
            if (mid < nums.back()) {
                r = m;
            } else {
                l = m + 1;
            }
        }
        
        return nums[l];
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1
        
        # array was not rotated
        if nums[l] <= nums[r]:
            return nums[l]
        
        while l < r:
            m = l + (r - l) // 2
            if nums[m] < nums[r]:
                r = m
            else:
                l = m + 1
                
        return nums[l]